### ISSUE TITLE
Y24-420 - As LeGo, we would like the new study set up fields to be added to the MLWH

### ISSUE THREAD

**Issue by harrietc52** [sanger/unified_warehouse#706]
**User story**
As LeGo, we would like the new study set up fields to be added to the MLWH, so that we can pull the data into eDAM and create DAA’s from it.

Currently, a script from eDAM pulls in the current study data from the MLWH study table at 2am everyday. It then uses this to create the DAA. They would like to pull in the other eDAM fields specified in Y24-419

When a Study in Sequencescape has been updated, the script in eDAM removes the existing studies in eDAM first, before inserting the studies obtained from MLWH, so all the studies in the system are up to date. However, DAAs that have already been approved do not get updated. This needs to be tested so all DAAs get updated

**For the final production system to be successful, it is important to ensure that Quan (Data Release Team) and LeGo are aware that they are responsible for checking that every new Study created in EBI, has a Legal Agreement (DAA) in eDAM. We are putting in place the restrictions so that all the EGA data must be added to Sequencescape on Study creation, and that the EGA data is present in the MLWH for eDAM to create the Legal agreement (DAA) using - But, it is up to them to check that there should be no releases to EGA without a legal agreement (DAA) in eDAM. Quan and LeGo should be involved in the testing of this story, ensuring they can view the new studies which are created and sequences uploaded. ** 

**Who are the primary contacts for this story**
@harrietc52 

**Who is the nominated tester for UAT**
Odion A

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] Give UAT MLWH RO credentials to eDAM devs for testing
- [ ] Put a story in eDAMs dev cycle, requesting the ability for them to pull in data from MLWH UAT
- [ ] Put a story in eDAMs dev cycle, requesting the ability for a study to be updated, after DAA is created and/or approved
- [ ] Unified Warehouse is updated to add the new Study fields to the `study` table in the MLWH
- [ ] On Study creation in SS, if the study contains the eDAM EGA fields (from Y24-419), add these fields to the MLWH `study` table 
- [ ] confirm the SQL query with Odion to enable eDAM to read the new fields added to the MLWH `study` table. (See current query used below)
- [ ] eDAM confirms access to all Study data required to generate a DAA
- [ ] eDAM only pulls in “standard” studies, whose data strategy is to be released in the EGA
- [ ] testing with eDAM to confirm when a study is created, a DAA is created
- [ ] testing with eDAM to confirm when a study is updated, an already created DAA is updated
- [ ] testing with eDAM to confirm when a study is updated, an already created and approved DAA is updated

**Dependencies**
This story is blocked by the following dependencies:
- sanger/sequencescape#4443 

**References**
This story has a non-blocking relationship with:
- Research story [Y24-374](https://github.com/sanger/sequencescape/issues/4407)

**Additional context**

Fields to be added:
[PSD Google Sheet with DAA fields](https://docs.google.com/spreadsheets/d/1Ya6YyBuoGNBV-6xFLfGHc1WuLjeVLjSlCOMQC6VCUhQ/edit?usp=sharing)

Current query used in eDAM to pull MLWH `study` data:
```
SELECT id_study_lims, description, accession_number, study_title, faculty_sponsor 
FROM study 
WHERE data_release_strategy=‘managed’
;
```

**Comment by harrietc52** [comment]
eDAM context:
- eDAM really old and bad
- eDAM2 - combine eDAM into eDAM2
- eDAM new build completed in ~3 months

Aim to go live with this work by end of year

**Comment by harrietc52** [comment]
Actions from meeting with Odion:
- go through Google sheet questions
- ask Ali H Google sheet questions
- meet with Ali and Odion to discuss timelines
- draw up Test Use Cases, which include alerts

**Comment by TWJW-SANGER** [comment]
Closing and archiving as this will be superseded by Sapio integration.
